src/library/scala/collection/immutable/LazyList.scala

```diff
@@ -518,15 +572,15 @@ final class LazyList[+A] private(private[this] var lazyState: () => LazyList.Sta
   override def prependedAll[B >: A](prefix: collection.IterableOnce[B]): LazyList[B] =
     if (knownIsEmpty) LazyList.from(prefix)
     else if (prefix.knownSize == 0) this
-    else newLL(stateFromIteratorConcatSuffix(prefix.iterator)(state))
+    else newLL(eagerHeadPrependIterator(prefix.iterator)(this))
 
   /** @inheritdoc
     *
     * $preservesLaziness
     */
   override def map[B](f: A => B): LazyList[B] =
-    if (knownIsEmpty) LazyList.empty
-    else (mapImpl(f): @inline)
```

Comment:
maybe this is actually needed  for heap safety / enabling early GC of `this`: https://github.com/scala/scala/pull/7990#issuecomment-524346566

Reply:
nah, was probably just me trying to prematurely optimise. `map` only evaluates a single element at a time, so it's not really an issue I don't think